exporters/metric/cortex/cortex.go

```diff
@@ -69,6 +93,231 @@ func InstallNewPipeline(config Config, options ...push.Option) (*push.Controller
 	return pusher, nil
 }
 
+// ConvertToTimeSeries converts a CheckpointSet to a slice of TimeSeries pointers
+func (e *Exporter) ConvertToTimeSeries(checkpointSet export.CheckpointSet) ([]*prompb.TimeSeries, error) {
+	var aggError error
+	var timeSeries []*prompb.TimeSeries
+
+	// Iterate over each record in the checkpoint set and convert to TimeSeries
+	aggError = checkpointSet.ForEach(e, func(record metric.Record) error {
+		// Convert based on aggregation type
+		agg := record.Aggregation()
+
+		// Check if aggregation has Sum value
+		if sum, ok := agg.(aggregation.Sum); ok {
```

Comment:
Would a [type switch](https://tour.golang.org/methods/16) be appropriate for handling the different aggregation types?

Reply:
Aggregation has a Kind value, but it's recommend to use conversions like this to handle user-defined aggregations.

From aggregation.go (sdk/export/metric/aggregation line 124-136):
```
	// Kind is a short name for the Aggregator that produces an
	// Aggregation, used for descriptive purpose only.  Kind is a
	// string to allow user-defined Aggregators.
	//
	// When deciding how to handle an Aggregation, Exporters are
	// encouraged to decide based on conversion to the above
	// interfaces based on strength, not on Kind value, when
	// deciding how to expose metric data.  This enables
	// user-supplied Aggregators to replace builtin Aggregators.
```